.ci/install_kata_image.sh

```diff
@@ -9,6 +9,17 @@ set -e
 
 cidir=$(dirname "$0")
 
+tmp_dir=$(mktemp -d -t kata-image-install.XXXXXXXXXX)
+readonly ROOTFS_DIR="${tmp_dir}/rootfs"
+export ROOTFS_DIR
+
+finish() {
+  sudo rm -rf "${ROOTFS_DIR}"
```

Comment:
This works, but it's worth pointing out that the *ordering* of the script is critical here.

Consider if the script looked like this (where the variables are set *after* the `trap`):

```
set -e

finish() {
  sudo rm -rf "${ROOTFS_DIR}"
}

trap finish EXIT

tmp_dir=$(mktemp -d -t kata-image-install.XXXXXXXXXX)
readonly ROOTFS_DIR="${tmp_dir}/rootfs"
export ROOTFS_DIR
```

Now, if `mktemp` fails, `finish()` *still* gets called on exit (even with `set -e`) and `ROOTFS_DIR=""` ! It might be prudent to check `ROOTFS_DIR` *just in case*:

```
finish() {
    [ -d "${ROOTFS_DIR}" ] && sudo rm -rf "${ROOTFS_DIR}"
}
```

/cc @grahamwhaley, @icarus-sparry, @chavafg.

Reply:
I suspect you are correct @jodh-intel - does look like a potential issue, and adding that check would be the correct thing do do imho

Reply:
@grahamwhaley @jodh-intel I already tested on bash does not not fail having an empty variable but I can add that extra validation, and because is removing with `sudo` (needed because the rootfs is owned by root) I can also double check that $ROOTFS contains `rootfs` just to make sure is not pointing to somewhere else.

Reply:
Historically things are even worse, when a pathname that was empty could be used to refer to the current directory. You can still see vestiges of this today, e.g.

     cd /   # make sure you are somewhere known 
     cd "" # empty name
     pwd  # should print /
     cd
     pwd  # should be your home directory

So yes, this is something to be concerned about. Essentially you need to write things in the correct order, and in particular don't set up traps which do things until after they are safe to be sprung. Just adding the test to see if ROOTFS_DIR is a directory isn't good enough, as the variable could be inherited from the environment, and you wouldn't like someone to pass say ROOTFS=/usr !

Of course `sudo` takes care to clean the environment to help for exactly this sort of reason.

The other point I would like to make is that traps are dynamic, you can set different things to do on exit (or error) at different points in the program. Of course if you take this too far then you have a spaghetti ball of unmaintainable code.